src/components/login/login.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
In the new version of react, declaring state object without necessarily placing it in the constructor is efficient

Reply:
@kodek-sleuth this is an inefficient reason, but I totally agree with you on a certain scale because the use of:
```
class  Login extends Component {
  state = {
    foo: bar
  }
}
```
is efficient when you are not passing props into the state. And that reason is inefficient when you are passing props as arguments in the state.